system/Database/Forge.php

```diff
@@ -1041,20 +1041,27 @@ protected function _processIndexes(string $table)
         return $sqls;
     }
 
-    protected function _processForeignKeys(string $table): string
+    protected function _processForeignKeys(string $table, array $allowActions = [], bool $onUpdate = true, bool $shortName = false): string
     {
         $sql = '';
 
-        $allowActions = [
-            'CASCADE',
-            'SET NULL',
-            'NO ACTION',
-            'RESTRICT',
-            'SET DEFAULT',
-        ];
+        if (empty($allowActions)) {
+            $allowActions = [
+                'CASCADE',
+                'SET NULL',
+                'NO ACTION',
+                'RESTRICT',
+                'SET DEFAULT',
+            ];
+        }
 
         foreach ($this->foreignKeys as $fkey) {
-            $nameIndex            = $table . '_' . implode('_', $fkey['field']) . '_foreign';
+            $nameIndex = $table . '_' . implode('_', $fkey['field']) . '_foreign';
+
+            if ($shortName === true && strlen($nameIndex) > 30) {
+                $nameIndex = substr($nameIndex, 0, 28) . mt_rand(10, 99);
+            }
```

Comment:
Why did you need to add this check? OCI8 does not show error if $nameIndex is too long?

This logic is only for OCI8. It should not be here.

Reply:
I moved it here and added $shortName parameter. This got rid of duplicate code. In OCI8 it calls parent:: with the $shortName if version less than 18

Reply:
In OCI8:

```php
    protected function _processForeignKeys(string $table, array $allowActions = [], bool $onUpdate = false, bool $shortName = false): string
    {
        $allowActions = ['CASCADE', 'SET NULL', 'NO ACTION'];

        if (version_compare($this->db->getVersion(), '18.0.0', '<')) {
            $shortName = true;
        }

        return parent::_processForeignKeys($table, $allowActions, $onUpdate, $shortName);
    }
```

Note: OCI8 does not have any actions on update either.

Reply:
Sorry, I don't get what you say.

The logic is obviously only for Oracle, so I think it should be in the OCI8 class only.

What if we want to add logic for MySQL that checks less than 65 characters?

Reply:
I suppose we can pass a parameter of max characters and apply it to all of them.

Reply:
Personally I hope an error occurs if the length is more than the database limit.